apps/mobile/src/components/HashDisplay/HashDisplay.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import { View, Text } from 'tamagui'
+import { TouchableOpacity } from 'react-native'
+import { Identicon } from '@/src/components/Identicon'
+import { EthAddress } from '@/src/components/EthAddress'
+import { SafeFontIcon } from '@/src/components/SafeFontIcon'
+import { useOpenExplorer } from '@/src/features/ConfirmTx/hooks/useOpenExplorer'
+import { Address } from '@/src/types/address'
+import { shortenAddress } from '@safe-global/utils/utils/formatters'
+import { CopyButton } from '@/src/components/CopyButton'
+import type { TextProps } from 'tamagui'
+
+export interface HashDisplayProps {
+  value: string | Address
+  showIdenticon?: boolean
+  showCopy?: boolean
+  showExternalLink?: boolean
+  identiconSize?: number
+  copyIconSize?: number
+  externalLinkSize?: number
+  textProps?: Partial<TextProps>
+  copyProps?: {
+    color?: string
+    size?: number
+  }
+  externalLinkColor?: string
+  /** Gap between elements */
+  gap?: string
+  onExternalLinkPress?: () => void
+  /** Whether to treat the value as an address (shows identicon) or generic hash */
+  isAddress?: boolean
```

Comment:
nit: It might be better to have both AddressDisplay and HashDisplay even if there is some duplication so its easier to extend each in the future without introducing more complexity. (I am just thinking about how much `EthHashInfo` has grown into a god component over the years).